binary_and_decimal.rb

```diff
@@ -12,6 +12,7 @@ def binary_to_decimal
       decimal_total += 2**i
```

Comment:
This will compute the decimal value incorrectly. If the binary number is 10000000 then the program will compute the decimal value to be 1 instead of 128. To correct it, you need to either reverse the binary_num before the loop, or compute the decimal_total as `decimal_total += 2**(7-i) if num == 1`